File: app/src/main/java/com/cpiekarski/fourteeners/utils/Mountain.java
package com.cpiekarski.fourteeners.utils;

public class Mountain {
    private String mName;
    private String mRange;
    private String mCounty;
    private int mRank;
    private int mElevation;
    private double mLongitude;
    private double mLatitude;
    
    private final String TAG = "Mountain";
    public static int NONE_OFFICIAL_RANK = -1;
    
    /**
     * 
     * @param name Name of the peak
     * @param range What range the peak is in
     * @param longitude Geo location
     * @param latitude Geo location
     * @param rank Arbitrary rank
     * @param elevation The peak elevation
     */
    Mountain(String name, String range, String county, double longitude, double latitude
            ,int rank
            ,int elevation ) {
        mName = name;
        mRange = range;
        mElevation = elevation;
        mRank = rank;
        mLongitude = longitude;
        mLatitude = latitude;
        mCounty = county;
    }
    
    public double getLongitude() {
        return mLongitude;
    }
    
    public double getLatitude() {
        return mLatitude;
    }
    
    public String getName() {
        return mName;
    }
    
    public String getRange() {
        return mRange;
    }
    
    public String getCounty() {
        return mCounty;
    }
    
    public int getElevation() {
        return mElevation;
    }
    
    @Override
    public String toString() {
        return "Name: " + mName + " Range: " + mRange + " Elevation: " + mElevation;
    }
}

File: app/src/main/kotlin/com/cpiekarski/fourteeners/utils/Mountain.kt
package com.cpiekarski.fourteeners.utils

data class Mountain(
    private val name: String,
    private val range: String,
    private val county: String,
    private val longitude: Double,
    private val latitude: Double,
    private val rank: Int,
    private val elevation: Int
) {
    companion object {
        const val NONE_OFFICIAL_RANK = -1
    }
    
    fun getLongitude(): Double = longitude
    fun getLatitude(): Double = latitude
    fun getName(): String = name
    fun getRange(): String = range
    fun getCounty(): String = county
    fun getElevation(): Int = elevation
    
    override fun toString(): String {
        return "Name: $name Range: $range Elevation: $elevation"
    }
}